src/server/scene/session_locker.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_SCENE_SESSION_HANDLER_H
+#define MIR_SCENE_SESSION_HANDLER_H
+
+#include "mir/frontend/session_locker.h"
+#include "mir/observer_multiplexer.h"
+#include <memory>
+
+namespace mf = mir::frontend;
+
+namespace mir
+{
+class Executor;
+class ConsoleServices;
+class MainLoop;
+
+namespace frontend
+{
+class SurfaceStack;
+class ScreenLockHandle;
+}
+
+namespace scene
+{
+
+class SessionLockerObserverMultiplexer : public ObserverMultiplexer<mf::SessionLockObserver>
+{
+public:
+    explicit SessionLockerObserverMultiplexer(std::shared_ptr<Executor> const& executor);
+    void on_lock() override;
+    void on_unlock() override;
+};
+
+class SessionLocker : public mf::SessionLocker
```

Comment:
I'm not convinced this is a class in it's own right. Certainly, it has a very intimate relationship with the `SurfaceStack`.

Reply:
I feel that the `SurfaceStack` can also be a `SessionLock`